File: 检测大写字母.cpp
//无脑
class Solution {
public:
    bool ifUpper(char c){
        if(c >= 'A' && c <= 'Z')
            return true;
        return false;
    }
    bool detectCapitalUse(string word) {
        bool ifShouldAllUpper = false;
        if(word.length() == 1)
            return true;
        if(!ifUpper(word[0]))
            ifShouldAllUpper = false;
        else if(ifUpper(word[0]) && !ifUpper(word[1]))
            ifShouldAllUpper = false;
        else if(ifUpper(word[0]) && ifUpper(word[1]))
            ifShouldAllUpper = true;
        for(int i = 1; i < word.length(); i++){
            if(ifUpper(word[i]) && !ifShouldAllUpper){
                return false;
            }
            if(!ifUpper(word[i]) && ifShouldAllUpper){
                return false;
            }
        }
        return true;
    }
};

File: 检测大写字母.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        n = len(word)
        if len(word) == 1:
            return True

        flag1 = False
        flag2 = True    
        for i in range(n):
            if i != 0 and word[i] >= 'A' and word[i] <= 'Z':
                flag1 = True
            
            if word[i] >= 'a' and word[i] <= 'z':
                flag2 = False
        
        if flag1 == True and flag2 == False:
            return False
        
        return True

